browser/android/background_video/background_video_playback_tab_helper.cc

```diff
@@ -65,15 +97,15 @@ BackgroundVideoPlaybackTabHelper::BackgroundVideoPlaybackTabHelper(
 
 BackgroundVideoPlaybackTabHelper::~BackgroundVideoPlaybackTabHelper() {}
 
-void BackgroundVideoPlaybackTabHelper::DidFinishNavigation(
-    content::NavigationHandle* navigation_handle) {
+void BackgroundVideoPlaybackTabHelper::PrimaryMainDocumentElementAvailable() {
+  content::WebContents* contents = web_contents();
   // Filter only YT domain here
-  if (!IsYouTubeDomain(web_contents()->GetLastCommittedURL())) {
+  if (!IsYouTubeDomain(contents->GetLastCommittedURL())) {
     return;
   }
-  if (IsBackgroundVideoPlaybackEnabled(web_contents())) {
-    web_contents()->GetPrimaryMainFrame()->ExecuteJavaScript(
-        kYoutubeBackgroundPlaybackScript, base::NullCallback());
+  if (IsBackgroundVideoPlaybackEnabled(contents)) {
+    contents->GetPrimaryMainFrame()->ExecuteJavaScript(
```

Comment:
we shouldn't be using `ExecuteJavaScript` here, it's only meant to be used in chrome:// and devtools urls. In a follow-up we should change this to use script injector component

Reply:
I may just do a follow-up to fix all of them